site/content/integrate/apps/quickstart/_index.md

```diff
@@ -1,18 +1,63 @@
 ---
-title: Quick start
-description: Guides for quickly getting started building apps
+title: Developer setup
+description: Developer setup and guides for quickly getting started building apps
 weight: 10
+aliases:
+  - /integrate/apps/quick-start-developer-setup
 ---
 
-The Apps quick start guides get you started developing your App in a variety of languages.
+To get started making Mattermost Apps, the first thing you need to do is set up your development environment via the [mattermost-app-examples repository](https://github.com/mattermost/mattermost-app-examples). Then you can follow language-specific guides to get started with the Apps Framework, or just start tinkering on your own.
+
+## Prerequisites
+
+You need [the latest version of Docker](https://docs.docker.com/get-docker/) installed.
+
+## Set up your development environment
+
+Clone and enter a local copy of the [mattermost-app-examples repository](https://github.com/mattermost/mattermost-app-examples) with the following command:
+
+```sh
+git clone https://github.com/mattermost/mattermost-app-examples && cd mattermost-app-examples
+```
+
+When you're developing your own app, you need an actual Mattermost server to be running. The development environment accomplishes this by using a [`docker-compose.yml`](https://github.com/mattermost/mattermost-app-examples/blob/master/docker-compose.yml) with a network named `mattermost-apps-dev`. This compose has just two containers: the Mattermost server with a Postgres database. The Mattermost image itself is preconfigured with the ideal settings for app development. 
+
+Because they exist on a pre-defined network, other `docker-compose.yml` configurations can connect their containers to this development environment by specifying the same `mattermost-apps-dev` Docker network. Then the Mattermost server can be accessed at `http://mattermost:8065` by any other container on the `mattermost-apps-dev` network. Similarly, your own development apps can then be accessed by Mattermost via their service name (e.g., `http://mattermost-apps-typescript-hello-world:4000`). You can learn more about Docker networks on the [official documentation](https://docs.docker.com/network/).
+
+To change the Mattermost Apps plugin or the Mattermost server versions, you can edit [`docker-compose.yml`](https://github.com/mattermost/mattermost-app-examples/blob/master/docker-compose.yml).
+
+Next, use the following command to bring your development environment online.
+
+```sh
+docker compose up -d
+```
+
+Once Mattermost is online, you can access the server at [http://localhost:8065](http://localhost:8065) and create your superuser account. You can also add `127.0.0.1 mattermost` to your `/etc/hosts` file to be able to access Mattermost at [http://mattermost:8065](http://mattermost:8065), which is the value of `MM_SERVICESETTINGS_SITEURL` for Docker networking.
```

Comment:
This is more like an extra tidbit. It does tell them they can access it at `localhost`, but in some cases you may want to be accessing Mattermost at the same URL that your Docker containers are. For example, image paths would be broken because everything in the server is mapped for `mattermost` as the host name, not `localhost`. I moved it to it's own paragraph and added **Tip:** to try and clarify this is just some extra insight.